code/modules/urist/modules/shipbattles/weapons.dm

```diff
@@ -282,6 +288,111 @@
+			to_chat(user, "<span class='notice'>You carefully undo the modifications to the warhead's circuitry.</span>")
+			riggedstate = CIRCUITRY_EXPOSED
+			playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+			icon_state = "torpedowarhead-open"
+		else if(riggedstate == CIRCUITRY_EXPOSED)
+			to_chat(user, "<span class='notice'>You begin to carefully modify the circuitry of the warhead.</span>")
+			if(do_after(user,4 SECONDS))
+				to_chat(user, "<span class='notice'>You have modified the torpedo warhead's internal circuitry. It can now be wired up and attached to something.</span>")
+				riggedstate = CIRCUITRY_MODIFIED
+				playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+				icon_state = "torpedowarhead-open-mod"
+		return
+	else if(istype(I, /obj/item/device/multitool))
+		if(!riggedstate)
+			return
+		to_chat(user, "<span class='notice'>You begin to [safety ? "disable" : "re-engage"] the built-in safeties.</span>")
+		if(do_after(user,4 SECONDS))
+			playsound(src.loc, 'sound/machines/buttonbeep.ogg', 25, 0, 10)
+			to_chat(user, "<span class='notice'>You [safety ? "disable" : "re-engage"] the built-in safeties.</span>")
+			safety = !safety
+			icon_state = "torpedowarhead-open-mod[safety ? "" : "-armed"]"
+	else if(istype(I, /obj/item/stack/cable_coil) && riggedstate == CIRCUITRY_MODIFIED)
+		to_chat(user, "<span class='notice'>You rig a wire from the torpedo warhead's detonator circuit. You can now attach something to it to detonate it remotely.</span>")
+		riggedstate = CIRCUITRY_RIGGED
+		playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+	else if(istype(I, /obj/item/device/assembly))
+		if(riggedstate == CIRCUITRY_RIGGED)
+			var/obj/item/device/assembly/A = I
+			to_chat(user, "<span class='warning'>You rig the [A] to the torpedo warhead's detonator circuit!</span>")
+			is_rigged = 1
+			if(!user.unEquip(A, src))
+				return
+			A.forceMove(src)
+			attached_device = I
+			A.holder = src
+			log_and_message_admins("[user] has rigged a torpedo IED.")
+			playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+			icon_state = "torpedowarhead-open-mod-armed"
+	return
+
+/obj/item/shipweapons/torpedo_warhead/proc/process_activation() // uh oh, time to boom
+	playsound(src.loc, 'sound/machines/buttonbeep.ogg', 25, 0, 10)
+	if(safety)
+		visible_message("<span class='danger'>[src] beeps stubbornly, refusing to detonate!</span>")
+		playsound(src.loc, 'sound/machines/buzz-sigh.ogg', 25, 0, 10)
+	if(!safety)
+		if(prob(15)) // Small chance for the warhead's safeties to engage briefly.
+			visible_message("<span class='danger'>[src] beeps stubbornly, refusing to detonate!</span>")
+			playsound(src.loc, 'sound/machines/buzz-sigh.ogg', 25, 0, 10)
+			return
+		visible_message("<span class='danger'>[src] pings, begining a short countdown!</span>")
+		playsound(src.loc, 'sound/machines/ping.ogg', 25, 0, 10)
+		playsound(src.loc, 'sound/items/countdown.ogg', 25, 0, 10)
+		spawn(4 SECONDS)
+			explosion(src, 2, 4, 6)
```

Comment:
I would maybe make this 1, 3, 5. That's about the only change I can think of though. Otherwise it seems okay, balance-wise. We'll have to see how it plays out in-round though.